assembly/src/ast/mod.rs

```diff
@@ -205,12 +208,26 @@ impl ProgramAst {
     // --------------------------------------------------------------------------------------------
 
     /// Returns byte representation of this [ProgramAst].
-    pub fn to_bytes(&self) -> Vec<u8> {
+    pub fn to_bytes(&self, options: AstSerdeOptions) -> Vec<u8> {
         let mut target = Vec::<u8>::default();
 
+        // serialize the options, so that deserialization knows what to do
+        options.write_into(&mut target);
+
         // asserts below are OK because we enforce limits on the number of procedure and the
         // number of body instructions in relevant parsers
 
+        if options.serialize_imports {
+            assert!(self.imports.len() <= MAX_IMPORTS, "too many imports");
+            target.write_u16(self.imports.len() as u16);
+            // We don't need to serialize the library names (the keys),
+            // since the libraty paths (the values) contain the library
+            // names
+            for (_, path) in self.imports.iter() {
+                path.write_into(&mut target);
+            }
```

Comment:
minor nit: we should be able to iterate over just values via `self.imports.values()` (we could use the same approach as on line 450 below).